### PR TITLE
LPS-168250 [Raylife AP] - Fix filters does not appear all options from Application

### PR DESCRIPTION
**Pull request by eduardo-diniz**
[LPS-168250](https://issues.liferay.com/browse/LPS-168250)

### CODE REVIEW

**Comment by eduardo-diniz** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx [hunk 1]
```diff
@@ -289,20 +299,30 @@ const ApplicationsTable = () => {
...
 						}
+					);
 
-						return filterList.push(item[propertyName]);
-					});
+					setProductFilterItems(products);
+				});
+
+				getPicklistByName('ApplicationStatus').then((results) => {
+					const policyStatusResult = results?.data?.listTypeEntries;
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx:316]:
```suggestion
				getPicklistByName('ApplicationStatus').then((results) => {
					const applicationStatusResult = results?.data?.listTypeEntries;
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx [hunk 2]
```diff
@@ -289,20 +299,30 @@ const ApplicationsTable = () => {
...
-				};
-				setProductFilterItems(filterApplications('productName'));
-				setStatusFilterItems(filterApplications('name'));
+					const policyStatuses = policyStatusResult.map(
+						(policyStatusPicklist: itemsPicklists) => {
+							return policyStatusPicklist.name;
+						}
+					);
+
+					setStatusFilterItems(policyStatuses);
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx:324]:
```suggestion
					const applicationStatuses = applicationStatusResult.map(
						(applicationStatusPicklist: itemsPicklists) => {
							return applicationStatusPicklist.name;
						}
					);

					setStatusFilterItems(applicationStatuses);
```

**Review by joanagrego — CHANGES_REQUESTED** [review]
Nice job! A added some suggestions :)

I saw that you also created the Picklists and Products files in the services. Be careful with rebase conflicts.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx (hunk 2 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx:324]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx (hunk 1 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx:316]: Valeu Joana!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx [hunk 3]
```diff
@@ -289,20 +299,31 @@ const ApplicationsTable = () => {
...
-				const filterApplications = (propertyName: string) => {
-					const filterList: string[] = [];
-					allItems.map((item: itemsApplications) => {
-						if (filterList.includes(item[propertyName])) {
-							return filterList;
+
+				getProducts().then((results) => {
+					const productsResult = results?.data?.items;
+
+					const products = productsResult.map(
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx:306]:
```suggestion
					const products = productsResult?.map(
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx [hunk 4]
```diff
@@ -289,20 +299,31 @@ const ApplicationsTable = () => {
...
+
+				getPicklistByName('ApplicationStatus').then((results) => {
+					const applicationStatusResult =
+						results?.data?.listTypeEntries;
 
-					return filterList;
-				};
-				setProductFilterItems(filterApplications('productName'));
-				setStatusFilterItems(filterApplications('name'));
+					const applicationStatuses = applicationStatusResult.map(
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx:319]:
```suggestion
					const applicationStatuses = applicationStatusResult?.map(
```

**Review by silasbsi — CHANGES_REQUESTED** [review]
I added some suggestions.

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:relevant

**Review by silasbsi — APPROVED** [review]

**Review by joanagrego — APPROVED** [review]
Approved! Thank you @eduardo-diniz :white_check_mark:

**Comment by silasbsi** [comment]
ci:forward

**Pull request closed without merging**